plutus-metatheory/src/Main.lagda

```diff
@@ -280,34 +288,48 @@ just t' ← withE runtimeError $ U.stepper maxsteps (ε ; [] ▻ erase t)
     where nothing → inj₁ (runtimeError userError)
     -}
   return $ prettyPrintUTm (U.extricateU0 (U.discharge V))
-executePLC TL t = do
+executePLC TL _ t = do
   (A ,, t) ← withE (λ e → typeError (uglyTypeError e)) $ typeCheckPLC t
   t' ← (withE runtimeError $ L.stepper t maxsteps) >>= checkError
   return (prettyPrintTm (unshifter Z (extricateScope (extricate t'))))
-executePLC TCK t = do
+executePLC TCK _ t = do
   (A ,, t) ← withE (λ e → typeError (uglyTypeError e)) $ typeCheckPLC t
   □ {t = t} v ← withE runtimeError $ Algorithmic.CK.stepper maxsteps (ε ▻ t)
     where ◆ _  → inj₁ (runtimeError userError)
           _    → inj₁ (runtimeError gasError)
   return (prettyPrintTm (unshifter Z (extricateScope (extricate t))))
-executePLC TCEK t = do
+executePLC TCEK _ t = do
   (A ,, t) ← withE (λ e → typeError (uglyTypeError e)) $ typeCheckPLC t
   □ V ← withE runtimeError $ Algorithmic.CEK.stepper maxsteps (ε ; [] ▻ t)
     where ◆ _  → inj₁ (runtimeError userError)
           _    → inj₁ (runtimeError gasError)
   return (prettyPrintTm (unshifter Z (extricateScope (extricate (Algorithmic.CEK.discharge V)))))
 
-executeUPLC : ⊥ U.⊢ → Either ERROR String
-executeUPLC t = do
-  □ V ← withE runtimeError $ U.stepper maxsteps (ε ; [] ▻ t)
-    where ◆  → inj₁ (runtimeError userError)
-          _    → inj₁ (runtimeError gasError)
-  return $ prettyPrintUTm (U.extricateU0 (U.discharge V))
 
-evalProgramNU : ProgramNU → Either ERROR String
-evalProgramNU namedprog = do
+showUPLCResult : U.State → Either ERROR String
+showUPLCResult (□ V) = return $ prettyPrintUTm (U.extricateU0 (U.discharge V))
+showUPLCResult ◆     = inj₁ (runtimeError userError)
+showUPLCResult _     = inj₁ (runtimeError gasError)
+
+executeUPLC : BudgetMode → ⊥ U.⊢ → Either ERROR String
+executeUPLC Silent t = (withE runtimeError $ U.stepper maxsteps (ε ; [] ▻ t)) >>= showUPLCResult
```

Comment:
You can't actually turn this on at the moment, right?  I think it's maybe doing the opposite to what `uplc` does: there it shows you the budget/tally without printing out the evaluation result.  I added that because sometimes I just wanted to see the cost or builtin usage of a program but it was also printing some massive term

Reply:
Actually, `Silent` here is meant to be the default option (not printing any budget, just the result). I should change the name to something else. I was not planning to implement `Silent` as in `uplc`, but maybe I should.

Reply:
Now I see where the confusion came from. In the implementation the constructor for "no cost printing" is called `Silent`, as I've done here. The option `--silent` is an *output* option (with constructor `NoOutput`). So I will leave this as it is so that constructors are similar to the implementation.